code/messaging/build.gradle

```diff
@@ -54,16 +58,37 @@ android {
         sourceCompatibility rootProject.ext.sourceCompatibility
         targetCompatibility rootProject.ext.targetCompatibility
     }
+
+    libraryVariants.all { variant ->
+        variant.getRuntimeConfiguration().exclude group: 'com.adobe.marketing.mobile', module: 'core'
+    }
+
+    sourceSets {
+        String sharedTestJavaDir = 'src/sharedTest/java'
+
+        test {
+            java.srcDirs += [sharedTestJavaDir]
+        }
+
+        androidTest {
+            java.srcDirs += [sharedTestJavaDir]
+            resources.srcDirs += ['src/test/resources']
+        }
+    }
 }
 
+def buildConfigDir = "$buildDir/generated/source/buildConfig/phone/debug/com/adobe/marketing/mobile/messaging/"
 android.libraryVariants.all { variant ->
     tasks.withType(Javadoc) {
-        source = [android.sourceSets.main.java.sourceFiles, android.sourceSets.phone.java.sourceFiles]
+        source = [android.sourceSets.main.java.sourceFiles,
+                  android.sourceSets.phone.java.sourceFiles]
+        source += buildConfigDir
```

Comment:
Do we need to do this? Since the buildconfigfields are part of defaultconfig, the BuildConfig.java should be included in all buildvariants.

Reply:
i think its safe to delete these